server/src/main/java/org/opensearch/wlm/ResourceType.java

```diff
@@ -24,19 +27,30 @@
  */
 @PublicApi(since = "2.17.0")
 public enum ResourceType {
-    CPU("cpu", task -> task.getTotalResourceUtilization(ResourceStats.CPU), true),
-    MEMORY("memory", task -> task.getTotalResourceUtilization(ResourceStats.MEMORY), true);
+    CPU("cpu", true, CpuUsageCalculator.INSTANCE, new ResourceUsageUtil() {
+        @Override
+        protected double getNormalisedThreshold(QueryGroup queryGroup) {
+            return queryGroup.getResourceLimits().get(ResourceType.CPU) * getSettings().getNodeLevelCpuCancellationThreshold();
+        }
+    }),
+    MEMORY("memory", true, MemoryUsageCalculator.INSTANCE, new ResourceUsageUtil() {
+        @Override
+        protected double getNormalisedThreshold(QueryGroup queryGroup) {
+            return queryGroup.getResourceLimits().get(ResourceType.MEMORY) * getSettings().getNodeLevelMemoryCancellationThreshold();
+        }
+    });
 
     private final String name;
-    private final Function<Task, Long> getResourceUsage;
     private final boolean statsEnabled;
-
+    private final ResourceUsageCalculator resourceUsageCalculator;
+    private final ResourceUsageUtil resourceUsageUtil;
```

Comment:
Can we rename this interface something other than Util?

Reply:
How about `QueryGroupUsage` since we are making decisons around current usage and the query group object